### DISCUSSION TITLE
create custom interface or any other?

### DISCUSSION THREAD

**Discussion by korneliusz-w** [directus/directus#6346] [Archive]
Hi

I want to create new type of field in interface.
I don't know how Vue is working, I'am Angular dev.

So my question are:
- do I have to create project in VueJS using VueCLI or only do this https://docs.directus.io/guides/interfaces/ ?
- how can I test my new field interface ?

EDIT: I'am tring to do this steps https://docs.directus.io/guides/interfaces/  but get an error on npx rollup -c

```
src/index.js → dist/index.js...
(!) Error when using sourcemap for reporting an error: Can't resolve original location of error.
src\interface.vue?vue&type=template&id=64969d30&lang.js (1:9)
[!] Error: 'openBlock' is not exported by ..\node_modules\vue\dist\vue.runtime.esm.js, imported by src\interface.vue?vue&type=template&id=64969d30&lang.js
https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module
src\interface.vue?vue&type=template&id=64969d30&lang.js (1:9)
1: import { openBlock as _openBlock, createBlock as _createBlock } from "vue"
```

**Comment by philleepflorence** [comment]
Linking related discussions, maybe one has a solution...
[Hotreloading Extensions](https://github.com/directus/directus/discussions/5938)
[Additional Node Modules for Extensions](https://github.com/directus/directus/discussions/5052)
[Make Custom Interfaces First Class Citizens](https://github.com/directus/directus/discussions/5386)

**Comment by rijkvanzanten** [comment]
> do I have to create project in VueJS using VueCLI or only do this https://docs.directus.io/guides/interfaces/ ?

Following the guide you linked is all you need 👍🏻 

> how can I test my new field interface ?

There's currently no dev-mode or HMR (ping @nickrum), so you'll have to follow the full guide and put it in your Directus' extensions folder 👍🏻

**Reply by philleepflorence** [reply]
So a local instance of Directus is required to test (deploy) interfaces, correct?
> Finally, move the output from your interface's dist folder into your project's `/extensions/interfaces/my-custom-interface` folder.

**Reply by rijkvanzanten** [reply]
Any instance will do, though a local one is arguably easier to work with (for quick restarting etc)

**Reply by philleepflorence** [reply]
WWRD - What Would Rijk Do...lol

**Reply by korneliusz-w** [reply]
I followed by instruction https://docs.directus.io/guides/interfaces/  and it is not working :(
I created rollup, moved index.js from /dist to /extensions/interfaces/custom/  , restart app but there is no new interface in field definition. I can see it in http://localhost:8055/extensions/interfaces/

The is a message in console:

Couldn't load custom interface "Custom": TypeError: Failed to resolve module specifier "vue". Relative references must start with either "/", "./", or "../".

??


EDIT: I found solution here https://www.gitmemory.com/issue/directus/directus/4401/803295907

downgrade rollup-plugin-vue  to 5.0.0 and install vue-template-compiler

**Comment by korneliusz-w** [comment]
Ups. The latest? version not working https://docs.directus.io/guides/interfaces/

```
- Building Directus extension...
(node:17920) UnhandledPromiseRejectionWarning: Error: Unexpected token (Note that you need plugins to import files that are not JavaScript)
    at error (D:\Projekty\directus-extensions\node_modules\rollup\dist\shared\rollup.js:151:30)
    at Module.error (D:\Projekty\directus-extensions\node_modules\rollup\dist\shared\rollup.js:9995:16)
    at Module.tryParse (D:\Projekty\directus-extensions\node_modules\rollup\dist\shared\rollup.js:10398:25)
    at Module.setSource (D:\Projekty\directus-extensions\node_modules\rollup\dist\shared\rollup.js:10301:24)
    at ModuleLoader.addModuleSource (D:\Projekty\directus-extensions\node_modules\rollup\dist\shared\rollup.js:19662:20)
    at async ModuleLoader.fetchModule (D:\Projekty\directus-extensions\node_modules\rollup\dist\shared\rollup.js:19718:9)
    at async Promise.all (index 2)
    at async ModuleLoader.fetchStaticDependencies (D:\Projekty\directus-extensions\node_modules\rollup\dist\shared\rollup.js:19744:34)
    at async Promise.all (index 0)
    at async ModuleLoader.fetchModule (D:\Projekty\directus-extensions\node_modules\rollup\dist\shared\rollup.js:19720:9)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:17920) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:17920) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by korneliusz-w** [comment]
Second problem. My custom interfaces which works in directus ^9.0.0-rc.69 stop working in ^9.0.0-rc.83

select-imageindex.js' is imported by virtual:entry, but could not be resolved – treating it as an external dependency
09:29:54 Loaded extensions: select-image
